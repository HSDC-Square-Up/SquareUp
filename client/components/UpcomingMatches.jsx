import React from "react";
import { useSelector } from "react-redux";

const UpcomingMatches = () => {
  const matches = useSelector(({ userSlice }) => userSlice.upcomingMatches);
  // console.log(matches);
  const matchElems = matches.map((element) => {
    // console.log("hello from forEach", element.name);
    return (
      <div className={"flex justify-between"} key={crypto.randomUUID()}>
        <div className="font-medium text-white">{element.name}</div>
        <div className="font-medium text-white">
          {element.totalWins}/{element.totalLosses}
        </div>
      </div>
    );
  });
  // console.log(matchElems);
  return (
    <div className="bg-dark-slate-green m-5 p-5 border-2 border-gray-700 overflow-scroll h-[90vh] rounded-lg">
      <div className="upcomingMatches">
        <h2 className="text-xl font-bold text-white bg-dark-slate-green p-2 px-4 sticky top-0 z-10">
          UPCOMING MATCHES
        </h2>
        <div id="wrapper" className={"flex flex-col"}>
          <div id="header-wrapper" className={"flex justify-between"}>
            <div className="font-extrabold text-white">Name</div>
            <div className="font-extrabold text-white">W/L</div>
          </div>
          <div id="matches" className={"flex flex-col"}>
            {matchElems}
          </div>
        </div>
      </div>
    </div>
  );
};

export default UpcomingMatches;
