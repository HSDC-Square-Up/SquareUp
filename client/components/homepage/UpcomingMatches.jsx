import React from 'react';

const UpcomingMatches = () => {

  
  return (
    <>
    <div className='upcomingMatches'>
      <h2>UPCOMING MATCHES</h2>
      <div id="upcomingMatchesContainer">
        <table>
          <thead>
            <tr>
              <th id="name">Name</th>
              <th id="winloss">W/L</th>
            </tr>
          </thead>

          <tbody id="leaderBoardBody">
          </tbody>

        </table>
      </div>
    </div>
    </>
  );
};

export default UpcomingMatches;
