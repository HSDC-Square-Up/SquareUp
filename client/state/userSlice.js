import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import client from '../lib/client.js';


export const loginUser = createAsyncThunk(
  'user/loginAsync',
  async(data, thunkAPI) => {
    try{
      const response = await client.post('/login', data, {});
      if(response.status = 200) {
        return response.data
      } else {
        return thunkAPI.rejectWithValue(`user/login: bad status: ${response.status}`);
      }
    } catch(err) {
      if (err.response && err.response.data.message) {
        return thunkAPI.rejectWithValue(err.response.data.message);
      } else {
        return thunkAPI.rejectWithValue(err.message);
      }
    }
})

export const getAllUsers = createAsyncThunk(
  'user/getAllUsersAsync',
  async(data, thunkAPI) => {
    try{
      const response = await client.get('/Leaderboard');
      if(response.status = 200) {
        return response.data
      } else {
        return thunkAPI.rejectWithValue(`user/getAllUsers: bad status: ${response.status}`);
      }
    } catch(err) {
      if (err.response && err.response.data.message) {
        return thunkAPI.rejectWithValue(err.response.data.message);
      } else {
        return thunkAPI.rejectWithValue(err.message);
      }
    }
})

const initialState = {
  user: {
    name: null,
    username: null,
    password: null,
    profilePicture: null,
    sex: null,
    height: null,
    weight: null,
    fightingStyle: null,
    totalWins: null,
    totalLosses: null,
  },
  userStatus: 'idle', // pending, success, failure
  users: [],
  allUsersStatus: 'idle',
  error: null,
  upcomingMatches: [],
};

export const userSlice = createSlice({
  name: 'user',
  initialState,
  reducers: {
    login: (state, action) => {
      state.user = Object.assign(state.user, action.payload);
    },
    getUsers: (state, action) => {
      state.users = action.payload;
    },
    addMatch: (state, action) => {
      state.users.push(action.payload);
    },
  },
  extraReducers: (builder) => {
    builder
      .addCase(loginUser.pending, (state, action) => {
        state.userStatus = 'pending';
      })
      .addCase(loginUser.fulfilled, (state, action) => {
        state.user = action.payload;
        state.userStatus = 'success';
      })
      .addCase(loginUser.rejected, (state, action) => {
        state.userStatus = 'failure';
        state.error = action.error.message;
      })
      .addCase(getAllUsers.pending, (state, action) => {
        state.allUsersStatus = 'pending';
      })
      .addCase(getAllUsers.fulfilled, (state,action) => {
         state.users = action.payload; 
         state.userStatus = 'success';
      })
      .addCase(getAllUsers.rejected, (state, action) => {
        state.allUsersStatus = 'failure'
        state.error = action.error.message;
      })
  }


});

export const { login, getUsers, addMatch } = userSlice.actions;

export default userSlice.reducer;
