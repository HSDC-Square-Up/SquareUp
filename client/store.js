import { configureStore } from '@reduxjs/toolkit';
import userSlice from './state/userSlice';

const store = configureStore({
  reducer: {
    userSlice,
  },
});

export default store;