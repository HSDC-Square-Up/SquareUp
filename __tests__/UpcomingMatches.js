import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import regeneratorRuntime from 'regenerator-runtime';
import { Provider, useDispatch } from 'react-redux';
import '@testing-library/jest-dom';
import { addMatch } from '../client/state/userSlice';

import store from '../client/store';
import UpcomingMatches from '../client/components/UpcomingMatches';

describe('Unit Testing Upcoming Matches', () => {
  let text;
  let testUser = {
    _id: '654d684da49332376cabe2a4',
    name: 'A#1 Cool Guy',
    username: 'coolestguy89',
    password: '$2a$10$N..IuiaA9lY20qcFshb22Oqz1KqhzAnvryOOXuDBYHQ6Fmz5FTTCu',
    profilePicture: 'none',
    wins: [],
    loss: [],
    totalWins: 0,
    totalLosses: 0,
    __v: 0,
    age: 33,
    fightingStyle: 'orange',
    height: 5,
    sex: 'male',
    weight: 5,
    location: 'Encinitas, CA'
  };

  beforeEach(() => {
    
    text = render(
      <Provider store={store} >
        <UpcomingMatches />
      </Provider>

    );
  });
  test('Has a header named upcoming matches', () => {
    expect(text.getByRole('heading')).toHaveTextContent('UPCOMING MATCHES');
  });
  test('Has a header-wrapper components', () => {
    expect(text.getByText('Name')).toBeInTheDocument()
    expect(text.getByText('W/L')).toBeInTheDocument()
  });
  // test('addMatch increases the number of matches displayed', () => {
  //   const dispatch = useDispatch();
  //   dispatch(addMatch(testUser));
  //   expect(text.getAllByText('A#1 Cool Guy').length).toBe(1)
  // })

});
