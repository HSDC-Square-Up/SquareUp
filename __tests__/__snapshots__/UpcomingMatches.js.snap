// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Testing Upcoming Matches should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-dark-slate-green m-5 p-5 border-2 border-gray-700 overflow-scroll h-[90vh] rounded-lg"
      >
        <div
          class="upcomingMatches"
        >
          <h2
            class="text-xl font-bold text-white bg-dark-slate-green p-2 px-4 sticky top-0 z-10"
          >
            UPCOMING MATCHES
          </h2>
          <div
            class="flex flex-col"
            id="wrapper"
          >
            <div
              class="flex justify-between"
              id="header-wrapper"
            >
              <div
                class="font-extrabold text-white"
              >
                Name
              </div>
              <div
                class="font-extrabold text-white"
              >
                W/L
              </div>
            </div>
            <div
              class="flex flex-col"
              id="matches"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-dark-slate-green m-5 p-5 border-2 border-gray-700 overflow-scroll h-[90vh] rounded-lg"
    >
      <div
        class="upcomingMatches"
      >
        <h2
          class="text-xl font-bold text-white bg-dark-slate-green p-2 px-4 sticky top-0 z-10"
        >
          UPCOMING MATCHES
        </h2>
        <div
          class="flex flex-col"
          id="wrapper"
        >
          <div
            class="flex justify-between"
            id="header-wrapper"
          >
            <div
              class="font-extrabold text-white"
            >
              Name
            </div>
            <div
              class="font-extrabold text-white"
            >
              W/L
            </div>
          </div>
          <div
            class="flex flex-col"
            id="matches"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
